keps/sig-node/2129-remove-cadvisor-json-metrics/README.md

```diff
@@ -0,0 +1,183 @@
+
+This was originally part of the ["metrics overhaul" KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/20181106-kubernetes-metrics-overhaul.md), but was [removed](https://github.com/kubernetes/enhancements/pull/1935) because it had not been completed. There were also [concerns raised](https://github.com/kubernetes/kubernetes/issues/68522#issuecomment-645454818) about removing these metrics.
+
+cAdvisor json metrics were disabled by default starting in 1.18, and could be re-enabled by setting `--enable-cadvisor-json-endpoints` to true.  However, given the concerns about removing the endpoints, we want to re-visit the deprecation and removal of these metrics before removing them permanently.
+
+## Motivation
+
+### Goals
+
+* Remove cAdvisor v1 ContainerInfo json metrics (`/stats/container`, `/stats/<podname>/<containername>`, `/stats/<namespace>/<podname>/<poduid>/<containername>`) from the kubelet.
+* Remove cAdvisor v1 MachineInfo json metrics (/spec) from the kubelet.
+
+### Non-Goals
+
+* Remove or modify cadvisor prometheus metrics from the kubelet (/metrics/prometheus).
+* Remove or modify the Summary API 
+* Eliminate the kubelet's dependence on cAdvisor for metrics to supply the Summary API.
+
+## Proposal
+
+### Risks and Mitigations
+
+The main risk that we face is breaking existing consumers of the cAdvisor json metrics.
+
+There are a few migration paths for users:
+
+For all metrics: Run cAdvisor as a daemonset.  See the [Instructions for running cAdvisor as a daemonset](https://github.com/google/cadvisor/tree/master/deploy/kubernetes#cadvisor-kubernetes-daemonset).
+* Pros: Provides the exact same APIs.
+* Cons: Can be expensive to run another instance of cAdvisor.
+
+For container metrics: Use an alternative kubelet endpoint.  Container metrics are available in /metrics/resource, /metrics/cadvisor, /stats/summary.
+* Pros: No additional cost
+* Cons: Metrics are in a different format and may not contain all information available in the json endpoints.
+
+For machine metrics: Use the prometheus node exporter.
+* Pros: Community-supported and widely used machine-level monitoring tool.  Easy-to-use configuration to enable/disable metrics.
+* Cons: Metrics are in a different format, and may not have the same set of information
+
+For most metrics, run a prometheus server, and collect metrics from the /metrics/cadvisor endpoint.
+If you want to be able to access metrics in JSON format, you can use the [Prometheus server's HTTP API](https://prometheus.io/docs/prometheus/latest/querying/api/).
+* Pros: Similar metrics in JSON format
+* Cons: The JSON structure of metrics is different, and it requires running a prometheus server.
+
+## Design Details
+
+Remove the `--enable-cadvisor-json-endpoints` flag and the kubelet stops serving on the paths listed in the Goals section.
+
+### Test Plan
+
+* This will not have any e2e testing.
+* There are no existing kubernetes e2e tests which check these endpoints.
```

Comment:
Just realized I commented on the resolved conversation, so re-commenting here. 

Test plans are required for all KEPs, if there is any test planned to be written (unit?), could you please include this here? Or are you suggesting that there will be no test at all?

Reply:
We can manually test this

Reply:
preferable, it would be great to have that written here. 
I can accept it into the release as it is since it's manual testing, but would it be possible to add that in here?

Reply:
@dashpole ^

Reply:
https://github.com/kubernetes/enhancements/pull/2525